src/nodes/composite/predicate.ts

```diff
@@ -264,12 +278,38 @@ export const assertAllowsConstraint = (
         case "morph":
             return
         default:
-            throwInternalError(`Unexpxected rule kind '${kind}'`)
+            throwInternalError(`Unexpxected rule kind '${node.kind}'`)
+    }
+}
+
+const assertValidLimit = (bounds: Range, boundType: "number" | "Date") => {
+    for (const index in bounds) {
+        const boundKind: BoundKind = index === "0" ? "right" : "left"
+        if (boundType === "number") {
+            if (typeof bounds[index].limit !== boundType) {
+                throwParseError(
+                    writeInvalidLimitMessage(
+                        bounds[index].comparator,
+                        `${bounds[index].limit}`,
+                        boundKind
+                    )
+                )
+            }
+        }
+        if (!isDate(bounds[index].limit) && boundType === "Date") {
+            throwParseError(
+                writeInvalidLimitMessage(
+                    bounds[index].comparator,
+                    `${bounds[index].limit}`,
+                    boundKind
+                )
+            )
+        }
     }
 }
 
 export const writeInvalidConstraintMessage = (
-    kind: ConstraintKind,
+    kind: string,
```

Comment:
Why did we change this lol? I vaguely remember it but what are we passing in that's not a constraint?

If you can change it back do, although if not it's not a big deal.